zipkin-common/src/test/scala/com/twitter/zipkin/storage/DependencyStoreSpec.scala

```diff
@@ -188,6 +188,51 @@ abstract class DependencyStoreSpec extends JUnitSuite with Matchers {
     result(store.getDependencies(today + 2 * day, Some(day))) should be(empty)
   }
 
+  @Test def getDependencies_intermediateSpans() = {
+    val day = MILLISECONDS.convert(1, DAYS)
+    val today = Time.now.floor(Duration.fromMilliseconds(day)).inMillis
+    val zipkinWeb = Endpoint(172 << 24 | 17 << 16 | 3, 8080, "zipkin-web")
+    val zipkinQuery = Endpoint(172 << 24 | 17 << 16 | 2, 9411, "zipkin-query")
+    val zipkinJdbc = Endpoint(172 << 24 | 17 << 16 | 2, 0, "zipkin-jdbc")
+
+    val trace = ApplyTimestampAndDuration(List(
+      Span(20L, "get", 20L,
+        annotations = List(
+          Annotation(today * 1000, Constants.ServerRecv, Some(zipkinWeb)),
+          Annotation((today + 350) * 1000, Constants.ServerSend, Some(zipkinWeb)))),
+      Span(20L, "call", 21L, Some(20L),
+        binaryAnnotations = List(
+          BinaryAnnotation(Constants.LocalComponent, "depth2", Some(zipkinWeb)))),
```

Comment:
I agree that capabilities API is not needed if you can simply ignore the test. Unfortunately, that's not always the case, e.g. the `headless_trace` test actually needs a different comparison target if the dependencies job can detect uninstrumented clients.

I also think introducing capabilities that require the actual business code doing if-then checks is generally a very bad idea, so I would carefully watch for that. The preferred way is when the store without the capability degrades gracefully into accepted default, e.g. in case of #930, the default null value (or `ParentAndChild` at the domain model) does not break anything and allows the rest of the code work as is.